### PR TITLE
Prefer imageio over PIL for file manipulations.

### LINKED ISSUE #2889: Feature request: skimage.io.imdecode
## Description
We have to write the image to disk for skimage to read from it.
Why not let skimage.io to decode form a bytes  string directly inside memory.

### PR DESCRIPTION
**Pull request by hmaarrfk**
Fixes #2889

Without `imageio` installed
```python
> ipython                                                                      (sk) 
Python 3.6.5 | packaged by conda-forge | (default, Apr  6 2018, 13:39:56) 
Type 'copyright', 'credits' or 'license' for more information
IPython 6.4.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import skimage.io

In [2]: skimage.io.reset_plugins()

In [3]: skimage.io.find_available_plugins(True)
Out[3]: 
{'pil': ['imread', 'imsave', 'imread_collection'],
 'matplotlib': ['imshow', 'imread', 'imshow_collection', 'imread_collection']}
```

After I installed `imageio`:
```python
> ipython                                                                      (sk) 
Python 3.6.5 | packaged by conda-forge | (default, Apr  6 2018, 13:39:56) 
Type 'copyright', 'credits' or 'license' for more information
IPython 6.4.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import skimage.io

In [2]: skimage.io.reset_plugins()

In [3]: skimage.io.find_available_plugins(True)
Out[3]: 
{'imageio': ['imread', 'imsave', 'imread_collection'],
 'matplotlib': ['imshow', 'imread', 'imshow_collection', 'imread_collection']}
```

We need to choose a minimum version. Here is the [release history](https://github.com/imageio/imageio/releases)

## Checklist
[It's fine to submit PRs which are a work in progress! But before they are merged, all PRs should provide:]
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Unit tests

## References
[If this is a bug-fix or enhancement, it closes issue # ]
[If this is a new feature, it implements the following paper: ]

## For reviewers

(Don't remove the checklist below.)

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

**Comment by hmaarrfk** [comment]
@stefanv  should I add `imageio` to the default requirements instead of `Pillow`?

File: requirements/default.txt [hunk 1]
```diff
@@ -2,7 +2,7 @@ numpy>=1.11
 scipy>=0.17.0
 matplotlib>=2.0.0
 networkx>=2.0
-pillow>=4.3.0
+imageio>=2.0
```
Comment (soupault) [inline requirements/default.txt]: This should be mentioned in `doc/release/release_dev.something`.

**Comment by hmaarrfk** [comment]
@jni can you restart this travis build please. I force removed a later commit and travis won't start on this.

**Comment by sciunto** [comment]
I'm +1 with the move. @hmaarrfk It needs a rebase please.

**Comment by hmaarrfk** [comment]
@sciunto I think the failures now are due to the pre build. I simplified the changes as many of the other ones were folded into #3231 that only makes modifications to the build scripts.

Feel free to squash merge if you wish.

**Comment by sciunto** [comment]
@scikit-image/core Opinion on this?

**Comment by blink1073** [comment]
👍 from me as well.

**Comment by sciunto** [comment]
@soupault I checked and nothing seems frankly different.

**Comment by hmaarrfk** [comment]
I just rebased and squashed to trigger a rebase.

**Comment by sciunto** [comment]
We have several positive comments, so I merge :) Thanks @hmaarrfk !

**Pull request merged**